### PR TITLE
feat: add mangopay connector

### PR DESCRIPTION
**Pull request by paul-nicolas**

### CODE REVIEW

File: components/payments/internal/app/connectors/mangopay/client/client.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	}
+
+	if err := c.login(context.Background()); err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
+
+func (c *Client) login(ctx context.Context) error {
```
Comment (gfyrag) [inline components/payments/internal/app/connectors/mangopay/client/client.go:59]: You can use : https://pkg.go.dev/golang.org/x/oauth2/clientcredentials

File: components/payments/internal/app/connectors/mangopay/task_fetch_users.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import (
+	"context"
+
+	"github.com/formancehq/payments/internal/app/connectors/mangopay/client"
+	"github.com/formancehq/payments/internal/app/models"
+	"github.com/formancehq/payments/internal/app/task"
+	"github.com/formancehq/stack/libs/go-libs/logging"
+)
+
+func taskFetchUsers(logger logging.Logger, client *client.Client) task.Task {
```
Comment (paul-nicolas) [inline components/payments/internal/app/connectors/mangopay/task_fetch_users.go:12]: Should be task_main

Comment (paul-nicolas) [reply components/payments/internal/app/connectors/mangopay/task_fetch_users.go:12]: Will be done in another PR

File: components/payments/internal/app/connectors/mangopay/client/client.go (hunk 1 above)
Comment (paul-nicolas) [reply components/payments/internal/app/connectors/mangopay/client/client.go]: Done thanks !

File: components/payments/internal/app/connectors/mangopay/client/transactions.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	}
+
+	defer func() {
+		err = resp.Body.Close()
+		if err != nil {
+			c.logger.Error(err)
+		}
+	}()
+
+	responseBody, err := io.ReadAll(resp.Body)
```
[the remark this answers is not in the document]
Comment (paul-nicolas) [reply components/payments/internal/app/connectors/mangopay/client/transactions.go]: Done thanks !

**Review by flemzord — APPROVED** [review]

**Review by gfyrag — APPROVED** [review]

**Pull request merged**